lib/rules/max-empty-lines/index.js

```diff
@@ -37,14 +37,15 @@ const rule = function(max, options) {
       return;
     }
 
-    const rootString = root.toString();
     const repeatLFNewLines = _.repeat("\n", maxAdjacentNewlines);
     const repeatCRLFNewLines = _.repeat("\r\n", maxAdjacentNewlines);
     const ignoreComments = optionsMatches(options, "ignore", "comments");
 
-    styleSearch({ source: rootString, target: "\n" }, match => {
-      checkMatch(rootString, match.endIndex, root);
-    });
+    if (root.constructor.name === "Document") {
```

Comment:
It's not a part of PostCSS, isn't it? Could you add a comment to the code explaining this `if` condition and what it `Document`, please?